stl/inc/deque

```diff
@@ -1236,52 +1322,78 @@ private:
         }
     };
 
-public:
-    template <class _Iter, enable_if_t<_Is_iterator_v<_Iter>, int> = 0>
-    iterator insert(const_iterator _Where, _Iter _First, _Iter _Last) {
-        // insert [_First, _Last) at _Where, input iterators
-        size_type _Off = static_cast<size_type>(_Where - begin());
+    template <bool _Bidi, class _It, class _Se>

```

Comment:
Hiss at boolean NTTP.

@AlexGuteniev can sing a song about my preference of enum class